### PR TITLE
Dynatrace.Observability merge to public repo

### PR DESCRIPTION
**Pull request by abhargavams**
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by abhargavams** [comment]
For avacado error, we have deprecated the API in the cleanup process. That API was not being used during the last stable version

image

**Comment by Wzb123456789** [comment]
Hi @divyansh3131 @dolauli please help check and approve this PR.

**Comment by abhargavams** [comment]
@Wzb123456789 

Please help us with the mentioned questions:

1. Avacado - I have mentioned the comment regarding that. I got approval when PR is merge into stable folder in private repo: [Features/users/abhargava/moveto stable folder by abhargavams · Pull Request #12287 · Azure/azure-rest-api-specs-pr (github.com)](https://github.com/Azure/azure-rest-api-specs-pr/pull/12287)
2. Want to understand about the API Readiness check.
3. What to do in swagger SDK breaking change tracking
4. Whom to take breaking change approval, we already got it in the PR mentioned in point 1

I tried to re-run the API readiness check but it's not getting triggered. Can you help with that?

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (sanar-microsoft) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:50]: >Dynatrace.Observability [](http://example.com/codeflow?start=82&length=23)

Just curious on the naming for this RP, it is different from other Microsoft.* named RPs, I've seen other Liftr RPs like datadog and elastic use Microsoft.* naming convention. Why is it different from them?

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (sanar-microsoft) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:902]: >VMSS [](http://example.com/codeflow?start=32&length=4)

Do you return VMSS info? From what I see response contains VM resource ID

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (sanar-microsoft) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:968]: >Monitors [](http://example.com/codeflow?start=40&length=8)

nit: Monitor

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (sanar-microsoft) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1262]: What does Status here mean? I see properties here include array of resource IDs but no status. Can you help me understand the scenario?

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/examples/Monitors_GetMetricStatus_MaximumSet_Gen.json
Comment (sanar-microsoft) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/examples/Monitors_GetMetricStatus_MaximumSet_Gen.json:14]: >enfghpfw [](http://example.com/codeflow?start=11&length=8)

Can you fix these and other new examples to have values closer to real values? Example: a string ARM resource ID format here?

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (visingla-ms) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1]: Please get a breaking changes approval from Azure Breaking Changes Reviewers <azbreakchangereview@microsoft.com> and ARM can review the changes afterwards. Once you have the approval, please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers.

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:50]: 2 years ago, when the project started PM suggested a new pattern for observability partners. Hence, they come up with Observability suffix

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1262]: The array is a list of azure resource ids that are sending metrics

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:968]: Made the changes

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/examples/Monitors_GetMetricStatus_MaximumSet_Gen.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/examples/Monitors_GetMetricStatus_MaximumSet_Gen.json:14]: provided the resource id in proper format

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1]: I have sent for the review for breaking change

**Comment by abhargavams** [comment]
@sanar-microsoft I have resolved the comments, please review.

**Comment by samsaha-ms** [comment]
@abhargavams, there are lots of cross version breaking change in swagger, either resolve them or get approval for breaking change for this PR for continuing ARM review further. https://aka.ms/AzBreakingChangesPolicy

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in https://github.com/Azure/azure-rest-api-specs-pr/pull/12287

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1]: @visingla-ms I have removed the ARMChangesRequested label as we got the breaking change approval.

**Comment by abhargavams** [comment]
@samsaha-ms We got the breaking change approval. Please do the needful.

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (samsaha-ms) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:947]: >/getMetricStatus" [](http://example.com/codeflow?start=128&length=17)

Just want to point out that creating such post actions does not work well with ARM echo system like ARM templates. Cant we model this as ARM resource?

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (samsaha-ms) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json]: Looks like it has not been addressed.

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (samsaha-ms) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1262]: Then is metrics status is proper naming here? It might confuse Cx. May be metrics emitters or something similar?

**Comment by samsaha-ms** [comment]
@abhargavams, please resolve the open comments in the PR and then remove "ARMChangesRequested" flag.

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json]: image

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1262]: @samsaha-ms For us, this array tells the status of resource which are emitting metrics. Can we keep it like that for our understanding?

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json]: The changes are there in PR

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:947]: We have previous post actions like this and hence following the same format. If such changes are required, then we would like to do at all places. Can we take it as a separate task in the future.

**Comment by abhargavams** [comment]
> @abhargavams, please resolve the open comments in the PR and then remove "ARMChangesRequested" flag.

Please review

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (samsaha-ms) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1262]: You should design APIs from Cx point of view too as they are the consumer for this APIs.

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (samsaha-ms) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:947]: That would be a breaking change too. Please the breaking change policy here https://aka.ms/AzBreakingChangesPolicy/

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (samsaha-ms) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:698]: >Get Marketplace SaaS resource details [](http://example.com/codeflow?start=24&length=37)

nit: As the api URI suggests that it would return the resource details of a tenant scoped under a specific subscription, may be update the description too to indicate that.

**Comment by samsaha-ms** [comment]
@abhargavams, once you resolve the open comments please remove ARMChangesRequested label.

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:1262]: This API we will be using internally and will ask not to publish to publicly.

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:947]: We already got approval while merging to private repo: https://github.com/Azure/azure-rest-api-specs-pr/pull/12287

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json]: image
Addressed

Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json]: image
Addressed

File: specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json
Comment (samsaha-ms) [inline specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json:698]: >specifc [](http://example.com/codeflow?start=82&length=7)

please fix the typo

Comment (abhargavams) [reply specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2023-04-27/dynatrace.json]: image
done

**Review by samsaha-ms — APPROVED** [review]
Since majority of the pr changes were already reviewed as part of https://github.com/Azure/azure-rest-api-specs-pr/pull/12287, signing off from ARM side.

**Review by samsaha-ms — APPROVED** [review]
Since majority of the pr changes were already reviewed as part of https://github.com/Azure/azure-rest-api-specs-pr/pull/12287, signing off from ARM side.

**Comment by Wzb123456789** [comment]
@samsaha-ms There is a `SuppressionReviewRequired` tag that limits merge, if Suppression is reviewed and approved, `Approved-suppression` label should be marked.

**Comment by samsaha-ms** [comment]
> @samsaha-ms There is a `SuppressionReviewRequired` tag that limits merge, if Suppression is reviewed and approved, `Approved-suppression` label should be marked.

@Wzb123456789 , please correct me if I am wrong the error is due to OperationIdNounVerb which SDK error as per https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/openapi-authoring-automated-guidelines.md#sdk-violations. Does the ARM team need to review and approve it?

Btw, @abhargavams please follow the process for approval https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process

**Comment by samsaha-ms** [comment]
> > @samsaha-ms There is a `SuppressionReviewRequired` tag that limits merge, if Suppression is reviewed and approved, `Approved-suppression` label should be marked.
> 
> @Wzb123456789 , please correct me if I am wrong the error is due to OperationIdNounVerb which SDK error as per https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/openapi-authoring-automated-guidelines.md#sdk-violations. Does the ARM reviewer team need to review and approve it?
> 
> Btw, @abhargavams you may choose to fix it or follow the process for suppression as it is an error since old api versions https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process

@abhargavams, in case you are choosing suppression path then tag the current idc devdiv arm on-call once you add the suppression in a comment with the justification for going with suppression, then the current idc devdiv on-call person (this week it's me) will do the rest of the needful.

**Comment by Wzb123456789** [comment]
@abhargavams  @samsaha-ms Perhaps a solution can be obtained from here. https://github.com/Azure/azure-rest-api-specs/pull/24101#issuecomment-1558357886

**Comment by samsaha-ms** [comment]
Posted here for the suppression review https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1685610617172?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1685610617172&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1685610617172

File: specification/dynatrace/resource-manager/readme.md
Comment (konrad-jamrozik) [inline specification/dynatrace/resource-manager/readme.md:43]: Can you clarify in the reason the API is deprecated? From [Teams](https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1685611811702?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1685610617172&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1685611811702) I've read:

> I guess the PR authors have added the suppression for this avocado rule -> MISSING_APIS_IN_DEFAULT_TAG, with the explanation that the API is deprecated and hence its cleaned up, its not being used in the last stable version.


Per the [TSG](https://github.com/Azure/avocado/blob/main/README.md#missing_apis_in_default_tag) we have:

> If the missing API swaggers are deprecated, in this case this error could be ignored.

So a reason that clarifies the API swagger is deprecated clear things up here.

**Comment by Wzb123456789** [comment]
@samsaha-ms Please check.

**Comment by samsaha-ms** [comment]
> @samsaha-ms Please check.

@Wzb123456789 The above comment by Konrad needs to be addressed by PR author @abhargavams.
@abhargavams, please work with the current ARM oncall person to get sign-off for suppression whenever you get the chance to fix it.

File: specification/dynatrace/resource-manager/readme.md
Comment (KasifMS) [reply specification/dynatrace/resource-manager/readme.md]: @konrad-jamrozik  - Can you check again? I see the readme.md is updated with reason now.  

@kanupriyasinghh @abhargavams

**Comment by konrad-jamrozik** [comment]
@abhargavams, @kanupriyasinghh

I see the Avocado suppression of `MISSING_APIS_IN_DEFAULT_TAG` is still not working, because I still see the error:

- https://github.com/Azure/azure-rest-api-specs/pull/24101/checks?check_run_id=14460884417

I see that in the commit [adding the suppression](https://github.com/Azure/azure-rest-api-specs/commit/6a4055b3cc011021c857d2ac2eef4f497538c94f) you are doing `from: swagger.yaml` which is wrong: `swagger.yml` is just a placeholder. You need to use the actual file.

Please refer to the guidance for suppressions here:
- https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/828/How-to-suppress-validation-rules?anchor=new-guidance-as-of-5/15/2023

Please note it has example of successful suppressions. For more examples, also see:
https://github.com/search?q=repo%3AAzure%2Fazure-rest-api-specs+suppress%3A+MISSING_APIS_IN_DEFAULT_TAG&type=code

Please let me know once you do a successful suppression, meaning the Avocado error won't show up anymore.

**Comment by kazrael2119** [comment]
this js breaking is caused by this pr.

File: specification/dynatrace/resource-manager/readme.md
Comment (qiaozha) [inline specification/dynatrace/resource-manager/readme.md:88]: I don't think we should add ruby and python is suppose to have a -track2 suffix. @msyyc

File: specification/dynatrace/resource-manager/readme.md
Comment (msyyc) [inline specification/dynatrace/resource-manager/readme.md:82]:
```suggestion
  - repo: azure-sdk-for-python-track2
```

**Comment by konrad-jamrozik** [comment]
[This suppression](https://github.com/Azure/azure-rest-api-specs/pull/24101/commits/e47350187e4f6534fa44cec8cb71539844793ee6) doesn't work because you need to use [the full path](https://github.com/Azure/azure-rest-api-specs/blob/a4e9dea863a8c4721e0bdf97b6ae8574ed9cd00f/specification/dynatrace/resource-manager/Dynatrace.Observability/stable/2021-09-01/dynatrace.json#L50C1-L50C1), like so:

`- $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Dynatrace.Observability/monitors/{monitorName}/getAccountCredentials"]`.

You can find more examples of this by [searching the repo](https://github.com/search?q=repo%3AAzure%2Fazure-rest-api-specs+%22where%3A+%24.paths%5B%5C%22%2Fsubscriptions%22&type=code)

**Comment by konrad-jamrozik** [comment]
@kanupriyasinghh I have removed Avocado suppressions because they do not work. I added `Approved-Avocado` label to make the Avocado check pass. Sorry for making you spend time on trying to make the suppressions work, my bad.

**Comment by kanupriyasinghh** [comment]
@rkmanda could you please merge the PR.

**Pull request merged**
